### DISCUSSION TITLE
Reason I want to add canonicalize to stablehlo

### DISCUSSION THREAD

**Discussion by linuxlonelyeagle** [openxla/community#106] [stablehlo]
There are stablehlo backend in [torch-mlir](https://github.com/llvm/torch-mlir), but they're not actually enabled right now.You can see this [PR](https://github.com/llvm/torch-mlir/pull/2712).We desperately want to add stablehlo's refbackend to torch-mlir.It doesn't look like there is an implementation of stablehlo.dynamic_reshape (with static shape) to linalg right now.So I think it can be added,or add canonicalize and it's dynamic_reshape to reshape and then to linalg.

**Comment by GleasonK** [comment]
Thanks for sharing! I've posted an RFC to OpenXLA Discuss which encompasses this proposal. This isn't the first instance I've seen of wanting some transformations on StableHLO for non-MHLO-based pathways, and I think moving _some_ of these transformations / canonicalizations to StableHLO actually makes for a cleaner overall picture.

[[RFC] Adding StableHLO transformations for CHLO, Shape, Canonicalizations](https://groups.google.com/a/openxla.org/g/openxla-discuss/c/httnobiya7U)

Let's move all discussion to that email chain as posts over there tend to have greater visibility.

**Comment by GleasonK** [comment]
To provide an update: General agreement on openxla-discuss posting. PRs have been reviewed and merged in https://github.com/openxla/stablehlo/pull/1984
